Lesson16/Lesson16-Busra/busra.js

```diff
@@ -0,0 +1,442 @@
+/*
+8. Print Each Character of a String (For-of)
+   - Define a function `printCharacters(str)` that uses a for-of loop
+     to log each character in the string on a separate line.
+*/
+
+function printCharacters(str) {
+  for (const char of str) {
+    console.log(char);
+  }
+}
+printCharacters("busra");
+
+
+/*
+9. Sum All Values in an Object
+   - Define a function `sumObjectValues(obj)` that iterates over the
+     properties of an object (using a for-in loop) and sums all numeric values.
+   - Log the sum.
+   - Example: {a: 10, b: 20, c: 5} -> 35
+*/
+const numbers={
+  num1: 10,
+  num2:50,
+  num3:3.5,
+  num4:4,
+  num5:"35"
+}
+let sum = 0;
+function sumObjectValues(obj){
+for(let num in obj){
+  if (typeof numbers[num] === "number"){
+    sum = sum + obj[num];
+  }
+  console.log(sum);
+}
+
+}
+
+sumObjectValues(numbers);
+
+
+
+/*
+10. Print Keys of an Object (For-in)
+    - Define a function `printObjectKeys(obj)` that uses a for-in loop
+      to log each key of the object.
+    - Example: { name: "Alice", age: 25 } -> logs "name", then "age"
+*/
+
+let person={
+ name:"Busra Demirkesen",
+ age:"34",
+ job:"Frontend Developer",
+ languages:{
+    Turkish:"Mother Language",
+    English:"B1+",
+    German:"A2",
+ },
+ skills:{
+    HTML:"good",
+    CSS:"good",
+    Javascript:"good"
+ }
+}
+
+
+function printObjectKeys(obj){
+  for(let key in obj){
+    if(typeof obj[key]==="object"&& !Array.isArray(obj[key])){ //I learned this structure by researching in AI because I was curious about what would happen if I used nested objects. Actually, I know that it's not necessary.
+      printObjectKeys(obj[key]);
+    }else{
+      console.log(`${key}: ${obj[key]}`);
+    }
+  }
+}
+
+printObjectKeys(person);
+/*
+11. Sum Array Using do-while Loop
+    - Define a function `sumWithDoWhile(numbers)` that uses a do-while loop
+      to sum all numbers in the array.
+    - Log the total.
+*/
+const numbersEx11 = [1,2,3,4,5,6,7,8,9,10];
+
+function sumWithDoWhile(arr){
+    let i=0;
+    let sumEx11 = 0;
+  do{
+    sumEx11 += arr[i];
+    i++;
+  }
+  while(i < arr.length){
```

Comment:
In the do...while loop, you don't need to add a set of `{}` after while. 
```suggestion
  while(i < arr.length);
```